stylegan_code_finder/lightning_modules/trans_u_net_lightning.py

```diff
@@ -26,13 +30,27 @@ def training_step(self, batch, batch_idx):
         return loss
 
     def validation_step(self, batch, batch_idx):
-        # this is the validation loop
+        super(TransUNetSegmenter, self).validation_step(batch, batch_idx)
         prediction = self.segmentation_network(batch['images'])
 
         ground_truth = torch.squeeze(batch['segmented'], dim=1)
         loss_ce = self.ce_loss(prediction, ground_truth.long())
         loss_dice = self.dice_loss(prediction, ground_truth, softmax=True)
         val_loss = 0.5 * loss_ce + 0.5 * loss_dice
-        self.log('dice_val_loss', loss_dice)
-        self.log('ce_val_loss', loss_ce)
-        self.log('val_loss', val_loss)
+        self.log('dice_val_loss', loss_dice, sync_dist=True)
+        self.log('ce_val_loss', loss_ce, sync_dist=True)
+        self.log('val_loss', val_loss, sync_dist=True)
+
+    def _initialize_segmentation_network(self):

```

Comment:
Would move this function to the top of the class (because it is done at the beginning)